viewport/viewport.go

```diff
@@ -123,6 +133,16 @@ func (m Model) visibleLines() (lines []string) {
 		bottom := clamp(m.YOffset+m.Height, top, len(m.lines))
 		lines = m.lines[top:bottom]
 	}
+
+	if m.indent > 0 {
+		cutLines := make([]string, len(lines))
+		for i := range lines {
+			cutLines[i] = ansi.TruncateLeft(lines[i], m.indent, "")
+		}
+
+		return cutLines
+	}
+
 	return lines
```

Comment:
I think this actually needs to be something like this:

```suggestion
	cutLines := make([]string, len(lines))
	for i, line := range lines {
		if m.indent > 0 {
			line = ansi.TruncateLeft(line, m.indent, "")
		}
		line = ansi.Truncate(line, m.Width, "")
		cutLines[i] = line
	}
	return cutLines
```

so it prints up to the width of the viewport only, otherwise long lines would still wrap to the next line

Reply:
current version:

https://github.com/user-attachments/assets/cede88e3-ac3b-4774-a3f9-94c139a7ce9d


with the proposed patch:

https://github.com/user-attachments/assets/9bc51051-af4b-4b0b-a4f2-c2910b421933

Reply:
This is way better. Thanks